diff --git a/node_modules/@medusajs/core-flows/dist/cart/workflows/create-payment-collection-for-cart.js b/node_modules/@medusajs/core-flows/dist/cart/workflows/create-payment-collection-for-cart.js
index 7173b5a..6fa84c3 100644
--- a/node_modules/@medusajs/core-flows/dist/cart/workflows/create-payment-collection-for-cart.js
+++ b/node_modules/@medusajs/core-flows/dist/cart/workflows/create-payment-collection-for-cart.js
@@ -84,7 +84,7 @@ exports.createPaymentCollectionForCartWorkflow = (0, workflows_sdk_1.createWorkf
     const paymentData = (0, workflows_sdk_1.transform)({ cart }, ({ cart }) => {
         return {
             currency_code: cart.currency_code,
-            amount: cart.raw_total,
+            amount: cart.total ?? cart.raw_total,
         };
     });
     const created = (0, create_payment_collection_1.createPaymentCollectionsStep)([paymentData]);
diff --git a/node_modules/@medusajs/core-flows/dist/cart/workflows/refresh-payment-collection.js b/node_modules/@medusajs/core-flows/dist/cart/workflows/refresh-payment-collection.js
index 5d8ea20..06af9fc 100644
--- a/node_modules/@medusajs/core-flows/dist/cart/workflows/refresh-payment-collection.js
+++ b/node_modules/@medusajs/core-flows/dist/cart/workflows/refresh-payment-collection.js
@@ -79,7 +79,8 @@ exports.refreshPaymentCollectionForCartWorkflow = (0, workflows_sdk_1.createWork
         cart,
     });
     (0, workflows_sdk_1.when)("should-update-payment-collection", { cart, shouldExecute }, ({ cart, shouldExecute }) => {
-        const valueIsEqual = utils_1.MathBN.eq(cart.payment_collection?.raw_amount ?? -1, cart.raw_total);
+        const cartTotal = cart.total ?? cart.raw_total;
+        const valueIsEqual = utils_1.MathBN.eq(cart.payment_collection?.raw_amount ?? -1, cartTotal);
         if (valueIsEqual) {
             return cart.payment_collection.currency_code !== cart.currency_code;
         }
@@ -99,7 +100,7 @@ exports.refreshPaymentCollectionForCartWorkflow = (0, workflows_sdk_1.createWork
             return {
                 selector: { id: cart.payment_collection.id },
                 update: {
-                    amount: cart.raw_total,
+                    amount: cart.total ?? cart.raw_total,
                     currency_code: cart.currency_code,
                 },
             };
